ui/components/multichain/nft-item/nft-item.js

```diff
@@ -16,27 +16,35 @@ import {
   JustifyContent,
 } from '../../../helpers/constants/design-system';
 import {
+  getCurrentNetwork,
   getIpfsGateway,
   getOpenSeaEnabled,
   getTestNetworkBackgroundColor,
 } from '../../../selectors';
+import useGetAssetImageUrl from '../../../hooks/useGetAssetImageUrl';
+import { getNftImageAlt } from '../../../helpers/utils/nfts';
 
-export const NftItem = ({
-  alt,
-  name,
-  src,
-  networkName,
-  networkSrc,
-  tokenId,
-  onClick,
-  clickable,
-  isIpfsURL,
-  badgeWrapperClassname,
-}) => {
+export const NftItem = ({ nft, onClick, clickable, badgeWrapperClassname }) => {
   const testNetworkBackgroundColor = useSelector(getTestNetworkBackgroundColor);
   const isIpfsEnabled = useSelector(getIpfsGateway);
   const openSeaEnabled = useSelector(getOpenSeaEnabled);
 
+  const { image, imageOriginal, name, tokenId } = nft;
+
+  const ipfsGateway = useSelector(getIpfsGateway);
+  const currentChain = useSelector(getCurrentNetwork);
+  const nftImageURL = useGetAssetImageUrl(imageOriginal ?? image, ipfsGateway);
+
+  const nftImageAlt = getNftImageAlt(nft);
+  const nftSrcUrl = imageOriginal ?? image;
+  const isIpfsURL = nftSrcUrl?.startsWith('ipfs:');
+  const isImageHosted = image?.startsWith('https:');
```

Comment:
No need for this change, but I typically prefer to use the [URL.protocol](https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol) property for this sort of thing.

Reply:
fixed [0912127](https://github.com/MetaMask/metamask-extension/pull/27724/commits/0912127050ba6e647f334f6be4588634e214d47d)

Reply:
Good point! However, using URL.protocol can throw an error if the string is not a valid URL. That's why I prefer to handle it safely by either validating the input or wrapping it in a try-catch to avoid potential issues with malformed URLs.